lib/buffer.js

```diff
@@ -105,6 +105,10 @@ function fromObject(obj) {
     return b;
   }
 
+  if (obj == null) {
```

Comment:
But `!obj` will give different error messages with Boolean values, as false value will enter this block.